odoo/addons/base/models/ir_cron.py

```diff
@@ -533,6 +627,43 @@ def _notifydb(self):
             cr.execute(query, [self.env.cr.dbname])
         _logger.debug("cron workers notified")
 
+    def _add_progress(self, *, timed_out_counter=0):
+        """ Create a progress record for the given cron and add it to its context.
+
+        :param int timed_out_counter: the number of times the cron has consecutively timed out
+        """
+        progress = self.env['ir.cron.progress'].create([{
+                'cron_id': self.id,
+                'remaining': 0,
+                'done': 0,
+                # we use timed_out_counter + 1 in order to detect when the
+                # current run times out
+                'timed_out_counter': timed_out_counter + 1,
+        }])
+        return self.with_context(ir_cron_progress_id=progress.id)
+
+    def _get_progress(self):
+        return self.env['ir.cron.progress'].browse(self.env.context.get('ir_cron_progress_id'))
+
+    def _notify_progress(self, done=1, remaining=None):
```

Comment:
In order to make calls clearer, we suggest
```suggestion
    def _notify_progress(self, *, done, remaining):
```
so that caller always have to pass `done=..., remaining=...`.